streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -80,7 +81,13 @@ void closeTask(final StreamTask task, final boolean clean) {
         }
     }
 
+    // Note that restoredPartitions may be nonempty even though restoring is, and will not be cleared in
+    // #updateRestored if this returns false, so we should clear any remaining partitions if restoring is empty
     boolean hasRestoringTasks() {
+        if (restoring.isEmpty()) {
```

Comment:
It seems a bit awkward to have a `hasXXX` call have side effects.  Although I do understand the motivation here, I guess this is something to revisit during a subsequent refactoring/follow-on PR.

Reply:
Yeah, that's fair (Guozhang said the same thing on the `trunk` PR 😄) -- I'll think of a better way to do this so that it's clear what's going on without having to comment and update the PR(s) by tonight

Reply:
Alright, I rewrote this clearing functionality to be an explicit method call and not a hidden side effect